framework/src/governance/lib/datazone/datazone-gsr-msk-datasource.ts

```diff
@@ -0,0 +1,232 @@
+
+  /**
+   * Build an instance of the DataZoneGsrMskDataSource
+   * @param scope the Scope of the CDK Construct
+   * @param id the ID of the CDK Construct
+   * @param props The DataZoneGsrMskDataSourceProps properties
+   */
+  constructor(scope: Construct, id: string, props: DataZoneGsrMskDataSourceProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: DataZoneGsrMskDataSource.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+
+    const stack = Stack.of(this);
+    const accountId = stack.account;
+    const region = stack.region;
+    const partition = stack.partition;
+
+    const clusterArn = `arn:${partition}:kafka:${region}:${accountId}:cluster/${props.clusterName}/*`;
+    const listClustersArn = `arn:${partition}:kafka:${region}:${accountId}:/api/v2/clusters`;
+    const glueRegistryArn = `arn:${partition}:glue:${region}:${accountId}:registry/${props.registryName}`;
+    const glueRegistrySchemasArn = `arn:${partition}:glue:${region}:${accountId}:schema/${props.registryName}/*`;
+
+    // Define SSM Parameter paths to store asset information
+    const parameterPrefix = `/datazone/${props.domainId}/${props.registryName}/asset/`;
+
+    this.datasourceLambdaRole = new Role(this, 'HandlerRole', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
+      ],
+      inlinePolicies: {
+        DataZonePermission: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: [
+                'datazone:CreateAsset',
+                'datazone:CreateAssetType',
+                'datazone:CreateFormType',
+                'datazone:GetAssetType',
+                'datazone:GetFormType',
+                'datazone:GetAsset',
+                'datazone:CreateAssetRevision',
+                'datazone:DeleteAsset',
+              ],
+              resources: [
+                `arn:${partition}:datazone:${region}:${accountId}:domain/${props.domainId}`,
+                `arn:${partition}:datazone:${region}:${accountId}:project/${props.projectId}`,
+              ],
+            }),
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: [
+                'glue:GetSchemaVersion',
+                'glue:ListSchemas',
+                'glue:ListSchemaVersions',
+              ],
+              resources: [glueRegistryArn, glueRegistrySchemasArn],
+            }),
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: [
+                'kafka:DescribeClusterV2',
+              ],
+              resources: [clusterArn],
+            }),
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: [
+                'kafka:ListClustersV2',
```

Comment:
this action does not take a resource, it requires a willdcard according to MSK [IAM reference](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonmanagedstreamingforapachekafka.html).

Reply:
We can split this policy into an independent one, so it has the * in the resources following the IAM reference